modules/openapi-generator/src/main/resources/kotlin-client/api.mustache

```diff
@@ -68,7 +68,7 @@ import {{packageName}}.infrastructure.toMultiValue
         )
 
         return when (localVarResponse.responseType) {
-            ResponseType.Success -> {{#returnType}}(localVarResponse as Success<*>).data as {{{returnType}}}{{/returnType}}{{^returnType}}Unit{{/returnType}}
+            ResponseType.Success -> {{#returnType}}(localVarResponse as Success<*>).data as {{{returnType}}}{{#nullableReturnType}}?{{/nullableReturnType}}{{/returnType}}{{^returnType}}Unit{{/returnType}}
```

Comment:
Shouldn't this be `as? Type` rather than `as Type?`? I've only used the first form and not the second.

See https://kotlinlang.org/docs/reference/typecasts.html#safe-nullable-cast-operator

Reply:
Both solutions are working. I would say the "cleaner" solution is to use `Type?` instead of `as?`. Have a look at the ""Unsafe" cast operator". It is true that `val x: String = y as String` crashes but `val x: String? = y as String?` would solve that. Does that makes sense to you?

Reply:
Yeah, I guess that's fine. As I understand it, `as T?` will throw if the cast fails while the other doesn't. If it becomes an issue, we could change it later.